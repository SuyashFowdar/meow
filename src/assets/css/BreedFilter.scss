.nav-container {
  max-width: 100vw;
  overflow-x: auto;
  box-shadow: 0 4px 5px rgba(0, 0, 0, 0.3);
  padding-top: 75px;
  background-color: rgb(50, 50, 50);

  & > div:first-child {
    margin-left: 10px;
    color: white;
  }

  .nav {
    color: #fafafa;
    padding: 5px 10px;
    height: 100%;
    white-space: nowrap;
    font-size: 16px;
  }

  .selected-nav,
  .nav:hover {
    color: white;
    font-weight: bold;
  }
}
