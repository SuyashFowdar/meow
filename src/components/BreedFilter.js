import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setFilter } from '../actions';
import '../assets/css/BreedFilter.scss';

const BreedFilter = ({
  handleSetFilter,
  breeds,
  category,
  setCategory,
}) => (
  <div className="row cross-center nav-container">
    <div>Breed: </div>
    {breeds.map((breed) => (
      <button type="button" className={`link nav ${category === breed.name ? 'selected-nav' : ''}`} key={Math.random()} onClick={() => { setCategory(breed.name); handleSetFilter(breed.name); }}>{breed.name}</button>
    ))}
  </div>
);

BreedFilter.propTypes = {
  handleSetFilter: PropTypes.func.isRequired,
  breeds: PropTypes.instanceOf(Array).isRequired,
  setCategory: PropTypes.func.isRequired,
  category: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => ({
  breeds: state.breeds,
});

const mapDispatchToProps = (dispatch) => ({
  handleSetFilter: (filter) => {
    dispatch(setFilter(filter));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(BreedFilter);
