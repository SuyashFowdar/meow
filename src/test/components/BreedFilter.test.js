import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import BreedFilter from '../../components/BreedFilter';
import reducer from '../../reducers/index';

it('renders correctly', () => {
  const store = createStore(reducer);
  const tree = renderer
    .create((
      <Provider store={store}>
        <BreedFilter category="All" setCategory={() => {}} />
      </Provider>
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
