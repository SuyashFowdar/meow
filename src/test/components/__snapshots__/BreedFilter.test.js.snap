// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row cross-center nav-container"
>
  <div>
    Breed: 
  </div>
  <button
    className="link nav selected-nav"
    onClick={[Function]}
    type="button"
  >
    All
  </button>
  <button
    className="link nav "
    onClick={[Function]}
    type="button"
  >
    Abyssinian
  </button>
  <button
    className="link nav "
    onClick={[Function]}
    type="button"
  >
    Aegean
  </button>
  <button
    className="link nav "
    onClick={[Function]}
    type="button"
  >
    American Shorthair
  </button>
  <button
    className="link nav "
    onClick={[Function]}
    type="button"
  >
    British Shorthair
  </button>
  <button
    className="link nav "
    onClick={[Function]}
    type="button"
  >
    Devon Rex
  </button>
  <button
    className="link nav "
    onClick={[Function]}
    type="button"
  >
    Exotic Shorthair
  </button>
  <button
    className="link nav "
    onClick={[Function]}
    type="button"
  >
    Maine Coon
  </button>
  <button
    className="link nav "
    onClick={[Function]}
    type="button"
  >
    Ragdoll
  </button>
  <button
    className="link nav "
    onClick={[Function]}
    type="button"
  >
    Scottish Fold
  </button>
  <button
    className="link nav "
    onClick={[Function]}
    type="button"
  >
    Persian
  </button>
</div>
`;
